openmdao_book/other/utils.py

```diff
@@ -0,0 +1,55 @@
+import json
+
+def reset_notebook(fname):
+    """
+    Empties the output fields and resets execution_count in all code cells in the given notebook.
+
+    Also removes any empty code cells. The specified notebook is overwritten.
+
+    Parameters
+    ----------
+    fname : str
+        Name of the notebook file.
+    """
+
+    with open(fname) as f:
+        dct = json.load(f)
+
+    newcells = []
+    for cell in dct['cells']:
+        if cell['cell_type'] == 'code':
+            if cell['source']:  # cell is not empty
+                cell['execution_count'] = None
+                cell['outputs'] = []
+                newcells.append(cell)
+        else:
+            newcells.append(cell)
+
+    dct['cells'] = newcells
+
+    with open(fname, 'w') as f:
+        json.dump(dct, f, indent=1)
+
+
+def reset_notebook_cmd():
+    import os, sys
+    if not len(sys.argv) == 2:
+        print(f"usage: python {os.path.basename(sys.argv[0])} <notebook_file>")
+        sys.exit(-1)
```

Comment:
would we want this to loop over arbitrary # of filenames so we could `reset_notebook *.ipynb`

Reply:
yeah that's a good idea

Reply:
done